darwin/dataset/utils.py

```diff
@@ -277,41 +278,55 @@ def get_coco_format_record(
         box_mode = BoxMode.XYXY_ABS
     except ImportError:
         box_mode = 0
-    data = attempt_decode(annotation_path)
-    height, width = data["image"]["height"], data["image"]["width"]
-    annotations = data["annotations"]
+    data = parse_darwin_json(annotation_path)
 
     record: Dict[str, Any] = {}
     if image_path is not None:
         record["file_name"] = str(image_path)
     if image_id is not None:
         record["image_id"] = image_id
-    record["height"] = height
-    record["width"] = width
+    record["height"] = data.image_height
+    record["width"] = data.image_width
 
     objs = []
-    for obj in annotations:
-        px, py = [], []
-        if annotation_type not in obj:
-            continue
+    for obj in data.annotations:
+        if annotation_type != obj.annotation_class.annotation_type:
+            if annotation_type not in obj.data:  # Allows training object detection with bboxes
+                continue
 
         if classes:
-            category = classes.index(obj["name"])
+            category = classes.index(obj.annotation_class.name)
         else:
-            category = obj["name"]
+            category = obj.annotation_class.name
         new_obj = {"bbox_mode": box_mode, "category_id": category, "iscrowd": 0}
 
         if annotation_type == "polygon":
-            for point in obj["polygon"]["path"]:
-                px.append(point["x"])
-                py.append(point["y"])
-            poly = [(x, y) for x, y in zip(px, py)]
-            if len(poly) < 3:  # Discard polyhons with less than 3 points
-                continue
-            new_obj["segmentation"] = [list(itertools.chain.from_iterable(poly))]
-            new_obj["bbox"] = [np.min(px), np.min(py), np.max(px), np.max(py)]
+            # Support for complex polygons
+            if "paths" in obj.data:
+                paths = obj.data["paths"]
+            elif "path" in obj.data:
+                paths = [obj.data["path"]]
+            else:
+                raise ValueError("polygon path not found")
+            all_px, all_py = [], []
+            segmentation = []
+
+            for path in paths:
+                px, py = [], []
+                for point in path:
+                    px.append(point["x"])
+                    py.append(point["y"])
+                poly = [(x, y) for x, y in zip(px, py)]
+                if len(poly) < 3:  # Discard polygons with less than 3 points
```

Comment:
This could be at the start of the function and it would skip any processing, `if len(path) < 3` rather than poly.